Client/src/Pages/Infrastructure/index.jsx

```diff
@@ -9,22 +71,228 @@ import { useNavigate } from "react-router-dom";
+	const { monitors, total: totalMonitors } = monitorState;
+	const monitorsAsRows = monitors.map((monitor) => ({
+		ip: monitor.name,
+		status: determineState(monitor),
+		processor: "2Ghz" /* How to retrieve that?  */,
+		cpu: 80 /* How to retrieve that?  */,
+		mem: 50 /* How to retrieve that?  */,
+		disk: 70 /* How to retrieve that?  */,
+	}));
+
 	return (
-		<Box component="main">
-			<Button
-				variant="contained"
-				color="primary"
-				onClick={() => {
-					navigate("/infrastructure/create");
+		<Stack
+			component="main"
+			style={{ width: "100%", gap: "1rem" }}
+		>
+			<Breadcrumbs list={[{ name: `infrastructure`, path: "/infrastructure" }]} />
+			<Stack
+				direction="row"
+				sx={{
+					justifyContent: "space-between",
+					alignItems: "center",
+					gap: "1rem",
+					flexWrap: "wrap",
+					marginBottom: "3rem",
+				}}
+			>
+				<Box style={{ maxWidth: "65ch" }}>
+					<Greeting type="uptime" />
+				</Box>
+				<Button
+					variant="contained"
+					color="primary"
+					onClick={navigateToCreate}
+					sx={{ fontWeight: 500 }}
+				>
+					Create infrastructure monitor
+				</Button>
+			</Stack>
+			<Stack
+				sx={{
+					gap: "1rem",
 				}}
-				sx={{ fontWeight: 500 }}
 			>
-				Create monitor
-			</Button>
-			<Heading component="h2">Infrastructure monitors</Heading>
-		</Box>
+				<Stack
+					direction="row"
+					sx={{
+						alignItems: "center",
+						gap: ".25rem",
+						flexWrap: "wrap",
+					}}
+				>
+					<Heading component="h2">Infrastructure monitors</Heading>
+					{/* TODO Correct the class current-monitors-counter, there are some unnecessary things there	 */}
+					<Box
+						component="span"
+						className="current-monitors-counter"
+						color={theme.palette.text.primary}
+						border={1}
+						borderColor={theme.palette.border.light}
+						backgroundColor={theme.palette.background.accent}
+					>
+						{totalMonitors}
+					</Box>
+				</Stack>
+				<TableContainer
+					component={Paper}
+					sx={{
+						maxHeight: "55vh",
+						"&::-webkit-scrollbar": {
+							width: ".75rem",
+						},
+						"&::-webkit-scrollbar-track": {
+							backgroundColor: theme.palette.secondary.light,
+						},
+						"&::-webkit-scrollbar-thumb": {
+							backgroundColor: theme.palette.secondary.contrastText,
+							outline: `1px solid ${theme.palette.secondary.contrastText}`,
+						},
+					}}
+				>
+					<Table stickyHeader>
+						<TableHead sx={{ backgroundColor: theme.palette.background.accent }}>
+							<TableRow>
+								{columns.map((column, index) => (
+									<TableCell
+										key={index}
+										align={index === 0 ? "left" : "center"}
+										sx={{
+											backgroundColor: theme.palette.background.accent,
+										}}
+									>
+										{column.label}
+									</TableCell>
+								))}
+							</TableRow>
+						</TableHead>
+						<TableBody>
+							{
+								/* ROWS */ monitorsAsRows.map((row, index) => (
+									<TableRow key={index}>
```

Comment:
We'll need an `onClick` listener for the row that opens the details page for that monitor